### ISSUE TITLE
DAPS token attribute referringConnector is ignored

### ISSUE THREAD

**Issue by trueg** [FraunhoferISST/DataspaceConnector#681]
Disclaimer: I am filing this as a bug rather than a feature request since it breaks security on white-listing contract offer policies.

The DAPS token contains the unique identifier for the connector certificate which is stored in `sub` and consists of a "combined entry of the SKI and AKI of the IDS X509" (see https://github.com/International-Data-Spaces-Association/IDS-G/blob/main/Components/IdentityProvider/DAPS/README.md). While this ID is unique it is not particularly useful for things like policy rules which require a URI to identify a connector. To that end the DAPS token contains the field `referringConnector` which should contain the URI/ID the connector claims in its self-information.

Now the provider connector should validate that this URI in `referringConnector` actually matches the one provided by the consumer connector, or at least use it for validating policy rules.

Currently `referringConnector` seems to be ignored and a consumer connector can claim any id they want. This makes policy rules like the one below essentially useless:

```
{
    "ids:constraint": [
      {
       "@type": "ids:Constraint",
       "@id": "https://w3id.org/idsa/autogen/constraint/e792d378-c15c-4d4e- 
    ae22-274ab6a8ac11",
       "ids:operator": {
        "@id": "https://w3id.org/idsa/code/SAME_AS"
       },
       "ids:leftOperand": {
        "@id": "https://w3id.org/idsa/code/SYSTEM"
       },
       "ids:rightOperand": {
        "@value": "https://my-consumer-connector.net/fancy",
        "@type": "http://www.w3.org/2001/XMLSchema#anyURI"
       }
      }
    ]
}
```

Here a connector is supposed to "own" the identifying URI `https://my-consumer-connector.net/fancy` in order to get access to the contract offer. This URI should be checked against the `referringConnector` in the DAPS token rather than the ID the connecting connector claims.

**Comment by juliapampus** [comment]
This check could be added as an additional processor as part of the message validation. Compare whether the issuer in the message header equals the one in the DAT. This will then be executed on every message. To not block test setups with custom certificates, do this check only in productive deployment.

**Comment by trueg** [comment]
production is fine, you could even make it configurable with default "false" to not break any existing setups?

**Comment by vdakker** [comment]
I implemented a check for this in the messaging-services (https://github.com/International-Data-Spaces-Association/IDS-Messaging-Services/pull/329).  It will be included in Version 5.0.2, so the DSC will get the fix when upgrading.
